README.md

```diff
@@ -4,4 +4,213 @@
   SPDX-License-Identifier: Apache-2.0
 -->
 
-# astarte-message-hub
+# Astarte Message Hub
+
+A central service that runs on (Linux) devices for collecting and delivering messages from N apps using 1 MQTT connection to Astarte.
+
+## Architecture
+The Astarte Message Hub is the main component that shares the Astarte connection to the nodes attached to it.
+The communication between the Hub and nodes is based on `gRPC` which is great for scenarios like real-time communication, low-power, low-bandwidth systems, and multi-language environments.
+`gRPC` makes the most out of `HTTP/2`, with multiplexed streaming and binary protocol framing. In addition, it offers performance advantages through the `Protobuf` message structure and features built-in code generation capability, which enables a multilingual environment.
```

Comment:
multilingual -> multi-language (I think the latter is more used for spoken languages)